core_info.c

```diff
@@ -620,7 +629,7 @@ static bool core_info_list_update_missing_firmware_internal(
    if (!core_info_list || !core)
       return false;
 
-   info                   = core_info_find_internal(core_info_list, core);
+   info                   = core_info_find_internal(core_info_list, core, NULL);
```

Comment:
Wrong number of arguments...?

Reply:
core_info_find() was a little strange: it accepted a core_info_ctx_find_t struct and a core_path parameter. 
core_info_ctx_find_t contains a inf member (where the found core data was written), and a core_path member (totally unused).
My impression here is that the original developer intended for core_info_ctx_find_t to contain both search parameters (core_path) and search results (inf). 
So i removed parameter core_path from core_info_find(), and modified it to use the core_path member in core_info_ctx_find_t, which was set before calling core_info_find(). 
I also added core_display_name to core_info_ctx_find_t in my inital implementation, now removed.

Reply:
You missed my point...

`core_info_find_internal()` accepts 2 arguments - here you have passed it 3. *This code will not compile*.